internal/app/marshaler.go

```diff
@@ -0,0 +1,40 @@
+package app
+
+import (
+	"google.golang.org/protobuf/encoding/protojson"
+	proto "google.golang.org/protobuf/proto"
+)
+
+// Marshaler describes anything that can marshal and unmarshal proto messages.
+type Marshaler interface {
+	Marshal(msg proto.Message) ([]byte, error)
+	Unmarshal(b []byte, msg proto.Message) error
+}
+
+// NewMarshaler returns a new marshaler.
+func NewMarshaler(json bool) Marshaler {
```

Comment:
ditto: no need to export this

Reply:
This is actually in the `app` package and I was thinking the client would use it too, so I think we prob need it to stay exported. Let me know though if you'd rather it stay in the `server` package.